src/main/java/frc/robot/subsystems/DriveSubsystem.java

```diff
@@ -0,0 +1,185 @@
+            .withCurrentLimits(35)
+            .coastInNeutral()
+            .withDirections(false, false)
+            .noMotorOutputLimits()
+            .noMotorStartupRamping()
+            .useSpeedControl()
+            .build();
+
+    private SparkMaxSettings positionSettings = SparkMaxSettingsBuilder.defaults()
+            .withCurrentLimits(35)
+            .coastInNeutral()
+            .withDirections(false, false)
+            .noMotorOutputLimits()
+            .noMotorStartupRamping()
+            .useSpeedControl()
+            .build();
+    private SparkMaxSettings smartMotionSettings = SparkMaxSettingsBuilder.defaults()
+            .withCurrentLimits(35)
+            .brakeInNeutral()
+            .withDirections(false, false)
+            .limitMotorOutputs(1.0, -1.0)
+            .noMotorStartupRamping()
+            .useSmartMotionControl()
+            .withPositionGains(RobotConstants.PID.DRIVE.F, 
+                RobotConstants.PID.DRIVE.P, 
+                RobotConstants.PID.DRIVE.I, 
+                RobotConstants.PID.DRIVE.D)
+            .useAccelerationStrategy(AccelStrategy.kTrapezoidal)
+            .withMaxVelocity(RobotConstants.AUTONOMOUS.MAX_VELOCITY)
+            .withMaxAcceleration(RobotConstants.AUTONOMOUS.MAX_ACCELLERATION)
+            .withClosedLoopError(RobotConstants.AUTONOMOUS.ACCEPTABLE_ERROR)
+            .build();
+
+    public Command reset() {
+        return new SingleShotCommand(this) {
+
+            @Override
+            public void doCommand() {
+                frontLeftPositionController.resetPosition();
+                frontRightPositionController.resetPosition();
+                rearLeftPositionController.resetPosition();
+                rearRightPositionController.resetPosition();
+                logger.log("Clicks per rotation", rearRightEncoder.getCountsPerRevolution());
+            }
+        }.withTimeout(EntechCommandBase.DEFAULT_TIMEOUT_SECONDS);
+
+    }
+
+    @Override
+    public void initialize() {
+        frontLeftSpark = new CANSparkMax(RobotConstants.CAN.FRONT_LEFT_MOTOR, MotorType.kBrushless);
+        rearLeftSpark = new CANSparkMax(RobotConstants.CAN.REAR_LEFT_MOTOR, MotorType.kBrushless);
+        leftSpeedController = new SpeedControllerGroup(frontLeftSpark, rearLeftSpark);
+
+        frontRightSpark = new CANSparkMax(RobotConstants.CAN.FRONT_RIGHT_MOTOR, MotorType.kBrushless);
+        rearRightSpark = new CANSparkMax(RobotConstants.CAN.REAR_RIGHT_MOTOR, MotorType.kBrushless);
+        rightSpeedController = new SpeedControllerGroup(frontRightSpark, rearRightSpark);
+
+        frontLeftEncoder = frontLeftSpark.getEncoder();
+        frontRightEncoder = frontRightSpark.getEncoder();
+        rearLeftEncoder = rearLeftSpark.getEncoder();
+        rearRightEncoder = rearRightSpark.getEncoder();
+
+        robotDrive = new DifferentialDrive(leftSpeedController, rightSpeedController);
```

Comment:
this code is duplicated with PositionDriveController.activate(). i think i like keeping all the motor/spark stuff here, and letting positiondrivecontroller simply manage position. The DriveSubsystem is the correct class to hold referneces to motors and motor controllers